dashboard/app/models/course_version.rb

```diff
@@ -18,4 +18,31 @@
 
 class CourseVersion < ApplicationRecord
   belongs_to :content_root, polymorphic: true
+
+  def self.update_course_version(content_root)
+    # TODO: Once the Course model is added, change this to just be version_year, since then the
+    # unique index will be on (course_id, key).
+    key = "#{content_root.family_name}-#{content_root.version_year}"
+
+    unless content_root.is_course?
+      CourseVersion.find_by(key: key)&.destroy
+      content_root.course_version = nil
+      content_root.save
+      return nil
+    end
+
+    course_version = CourseVersion.find_or_create_by(
+      key: key,
+      display_name: content_root.version_year,
+      content_root: content_root,
+    )
+
+    # Delete old CourseVersion if the key has been changed to something else
+    content_root.course_version.destroy if course_version != content_root.course_version
+    content_root.course_version = course_version
+
+    # TODO: add relevant properties from content root to course_version
```

Comment:
What does this mean exactly -- are we talking about the "delegate" syntax from earlier?

Reply:
For the short term, yes, and for the long term, to move the properties to actually be stored on this model.